### ISSUE TITLE
Passing spider initialization options 

### ISSUE THREAD

**Issue by Ziinc** [oltarasenko/crawly#126]
`init/0` results in a "hardcoded" way of providing initial urls to crawl.

For example, during application runtime, if I wanted to feed urls to the spider to crawl, I would not be able to do so without having a convoluted method of fetching those urls (e.g. from a database).

I propose allowing options to be passed as arguments to the spider's `init` callback, passed through the `Crawly.Engine.start_spider` function. These options are optional, and is up to the user to implement it in the spider.

Example:
```elixir
Crawly.Engine.start_spider(MySpider, urls: ["my urls"], pagination: false)

#in the spider
def init(opts \\ []) do
  # default options
  opts = Enum.into(opts, %{urls: ["Other url"], pagination: true})
  # do something with pagination flag
  # ....

  [start_urls: opts[:urls] ]
end
```

**Comment by oltarasenko** [comment]
@Ziinc Yes we can do it. I can take a look on it. I am now working on automatic spiders generation and did something similar.. I any case it seems it will be fast to build it.

**Comment by Ziinc** [comment]
@oltarasenko spider codegen or dynamic spider definitions? I've got a few possible use cases for dynamic spiders, it would be an interesting area to explore.

I think startup options passed through the engines should be quite simple to implement, I'll make a PR this week

**Comment by oltarasenko** [comment]
I have quite an interesting POC running as a part of CrawlyUI (http://crawlyui.com/). The idea is to allow generating the spider through the web. The backbone for it works already, so I was able to write one of the simple spiders. Now I am working on the interface. Will try to publish this work soon.

**Comment by oltarasenko** [comment]
Released in 0.12.0
